pkg/util/sys/device.go

```diff
@@ -91,6 +102,60 @@ func ListDevices(executor exec.Executor) ([]string, error) {
 	return strings.Split(devices, "\n"), nil
 }
 
+func GetCephVolumeInventory(executor exec.Executor) (*[]CephVolumeDeviceData, error) {
+	cmd := "ceph-volume inventory --format plain"
+	inventory, err := executor.ExecuteCommandWithOutput(false, cmd, "ceph-volume", "inventory", "--format", "json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ceph-volume inventory: %+v", err)
+	}
+
+	// Slice to store device properties
+	CVStorageDevices := []CephVolumeDeviceData{}
+
+	// Get a slice to store the json data
+	bInventory := []byte(inventory)
+	var f interface{}
+	err = json.Unmarshal(bInventory, &f)
+	if err != nil {
+		return &CVStorageDevices, nil
+	}
+	deviceItems := f.([]interface{})
+
+	// Get the data from the different devices
+	for _, item := range deviceItems {
+		deviceData := item.(map[string]interface{})
+		if _, exist := deviceData["sys_api"]; exist {
+			deviceDetails := deviceData["sys_api"].(map[string]interface{})
+
+			newCVDevice := CephVolumeDeviceData{
+				DevicePath: getStringAttributeFromMapInterface(deviceData, "path"),
+				Size:       getStringAttributeFromMapInterface(deviceDetails, "human_readable_size"),
+				Rotates:    getStringAttributeFromMapInterface(deviceDetails, "rotational") == "1",
+				Available:  getBoolAttributeFromMapInterface(deviceData, "available"),
+				ModelName:  getStringAttributeFromMapInterface(deviceDetails, "model")}
+
+			CVStorageDevices = append(CVStorageDevices, newCVDevice)
+		}
+	}
+	return &CVStorageDevices, nil
+}
+
+func getBoolAttributeFromMapInterface(sourceMap map[string]interface{}, attribute string) bool {
+	if _, exist := sourceMap[attribute].(bool); exist {
+		return sourceMap[attribute].(bool)
+	} else {
```

Comment:
You don't need the `else`, just `return false` at the end of the function.

Reply:
thx!